// gitbook-plugin-theme-fs/src/js/theme/imgpreview.js
var $ = require('jquery');

var platform = require('./platform');

function init() {

    // 初始化微信JSSDK
    if (platform.isWXBrowser()) {
        $.getScript('https://res.wx.qq.com/open/js/jweixin-1.2.0.js', function(resp) {
            if (!window.wx) {
                return;
            }
            window.wx.ready(function(resp) {
                console.log(resp)
            });
        })
    }

    // 初始化JSAPI
    if (platform.isFSBrowser()) {
        $.getScript('https://www.fxiaoke.com/open/jsapi/2.1.5/fsapi.min.js', function(resp) {
            if (!window.FSOpen) {
                return;
            }
            window.FSOpen.init({
                appId: 'FSAID_5f5e533',
                internalId: false,
                onSuccess: function(resp) {
                    console.log(resp)
                },
                onFail: function(error) {
                    console.warn(error)
                }
            });
        })
    }

    $(document).on('click', '.book-body .markdown-section img', function(e) {
        var $target = $(e.target),
            _src = $target.attr('src');

        var link = document.createElement('a');
        link.href = _src;

        if (window.FSOpen) {
            window.FSOpen.media.image.preview({
                index: 0,
                imgUrls: [ link.href ]
            });
        } else if (window.wx) {
            window.wx.previewImage({
                current: link.href,
                urls: [ link.href ]
            });
        } else {
            window.open(_src);
        }
    });
}

module.exports = {
    init: init
}
